ballerina/auth_utils.bal

```diff
@@ -48,14 +48,22 @@ isolated function buildCompleteErrorMessage(error err) returns string {
 // Extract the credential from `http:Request`, `http:Headers` or `string` header.
 isolated function extractCredential(Request|Headers|string data) returns string|ListenerAuthError {
     if data is string {
-        return re`\s`.split(data)[1];
+        string[] parts = re`\s`.split(data);
+        if parts.length() < 2 {
+            return prepareListenerAuthError("Invalid authorization header format.");
+        }
+        return parts[1];
     } else {
         object {
             public isolated function getHeader(string headerName) returns string|HeaderNotFoundError;
         } headers = data;
         var header = headers.getHeader(AUTH_HEADER);
         if header is string {
-            return re`\s`.split(header)[1];
+            string[] parts = re`\s`.split(header);
+            if parts.length() < 2 {
+                return prepareListenerAuthError("Invalid authorization header format.");
+            }
+            return parts[1];
```

Comment:
I think we can extract the splitting and validating part to a separate function since it is duplicated?

Reply:
Fixed in [9a427e1](https://github.com/ballerina-platform/module-ballerina-http/pull/2573/commits/9a427e17c92c0e8994da9b687f7bc2b50603973d)